vk/backend/backend.go

```diff
@@ -113,169 +116,302 @@ func state2containerState(prevState *v1.ContainerState, currState titusdriver.Ti
 	}
 }
 
-func RunWithBackend(ctx context.Context, rp runtimeTypes.ContainerRuntimeProvider, m metrics.Reporter, statuses *os.File, pod *v1.Pod, cfg config.Config) error { // nolint: gocyclo
+type Backend struct {
```

Comment:
In general, it's better to not expose a struct like this because it could be accidentally instantiated by an external package incorrectly. It's better to expose an interface.